File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        n = 0
        res = []
        for c in s:
            if c=='(':
                if n>0:
                    res.append(c)
                n+=1
            else:
                if n > 1:
                    res.append(c)
                n -= 1
        return "".join(res)

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string res;
        int n = 0;
        for(char c:s){
            if (c == '(' && n++ >0) res +=c;
            else{
                if(c==')' && n-- >1) res +=c;
            }
        }
        return res;
    }
};